lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -223,9 +223,22 @@ def add_payment_token(source, payment, options)
         end
       end
 
+      def add_decypted_wallet(source, payment, options)
+        source[:sourceType] = 'DecryptedWallet'
+        source[:card] = {
+          cardData: payment.number,
+          expirationMonth: (format(payment.month, :two_digits) if payment.month),
+          expirationYear: (format(payment.year, :four_digits) if payment.year)
+        }
+        source[:cavv] = payment.payment_cryptogram
+        source[:walletType] = payment.source.to_s.upcase if payment.source == :apple_pay || payment.source == :google_pay
```

Comment:
💬 Comment:

Same case here if `payment.source` is not `:apple_pay` or `:google_pay` you end-up sending an incomplete object.

Reply:
I think that even the validation is unnecessary, by default the source will be apple_pay if no source match with the listed 

[sources](https://github.com/activemerchant/active_merchant/blob/ee0806ff7aefa4998e709d270dd123838a4bee2b/lib/active_merchant/billing/network_tokenization_credit_card.rb#L20)